design/KWOK.md

```diff
@@ -0,0 +1,45 @@
+# Support running on KWOK cluster
+
+## Ticket
+https://runai.atlassian.net/browse/RUN-16744
+
+## Motivation
+The KWOK cluster enables running fake pods on fake nodes, offering an opportunity to reduce the cost of running scale tests on real nodes. 
+This initiative aims to extend support for our system to operate seamlessly within the KWOK cluster environment, so we can reduce the cost of running scale tests on real nodes.
+
+## Assumptions
+- Pods running on fake nodes are considered fake pods. These pods' containers will not execute.
+
+## Requirements
+- Ensure labels and metrics are exposed in the same manner as on real nodes.
+
+## Gaps
+- Status Exporter
+  - The current deployment as a DaemonSet is incompatible with fake nodes.
+- RunAI's Node Exporter
+  - The current deployment as a DaemonSet is incompatible with fake nodes.
+- Device Plugin
+  - The current deployment as a DaemonSet is incompatible with fake nodes. We might want to not supoprt it on fake nodes and require manual node resources update.
```

Comment:
why is this different than the others? what is the manner of the manual change you mention here?

Reply:
Since it is easier to replace manually.
Replacing it manually means editing the capacity and allocatable sections of nodes.
Replacing the status exporter will require exporting metrics and labels manually which is harder.